pkg/controller/elasticsearch/driver/driver.go

```diff
@@ -321,6 +321,15 @@ func (d *defaultDriver) newElasticsearchClient(
 	)
 }
 
+// isReachable will check if Elasticsearch has any pods ready to handle requests.
+func (d *defaultDriver) isReachable() (bool, error) {
+	resourcesState, err := reconcile.NewResourcesStateFromAPI(d.Client, d.ES)
+	if err != nil {
+		return false, err
+	}
+	return len(resourcesState.CurrentPodsByPhase[corev1.PodPhase(corev1.PodReady)]) > 0, nil
```

Comment:
This won't work `Ready` is a Pod condition and not a phase.

Reply:
Good point.  I've updated this to take into account both the phase of the pod, and the conditions.  There's also a set of tests around this behavior.